typescript/sdk/src/metadata/agentConfig.ts

```diff
@@ -138,25 +163,47 @@ export const AgentChainMetadataSchema = ChainMetadataSchemaObject.merge(
 
     switch (metadata.protocol) {
       case ProtocolType.Ethereum:
-        return [
-          AgentSignerKeyType.Hex,
-          signerType === AgentSignerKeyType.Aws,
-          signerType === AgentSignerKeyType.Node,
-        ].includes(signerType);
+        if (
+          ![
+            AgentSignerKeyType.Hex,
+            signerType === AgentSignerKeyType.Aws,
+            signerType === AgentSignerKeyType.Node,
+          ].includes(signerType)
+        ) {
+          return false;
+        }
+        break;
```

Comment:
Why do people keep using switch statements? The worst control flow tool ever invented haha

Reply:
hahaha fair point